client-js/app/elements/labrad-plot.ts

```diff
@@ -351,14 +584,113 @@ export class Plot extends polymer.Base {
       this.limits.yMin -= 1;
       this.limits.yMax += 1;
     }
+  }
 
-    // Zoom to fit the data.
-    // TODO: Only if the zoom has not changed.
-    this.resetZoom();
+
+  /**
+   * Generate the geometries and materials for a 1D plot.
+   */
+  private plotData1D(data: number[][]) {
+    this.dataLimits1D(data);
+
+    const ob = new THREE.Object3D();
+
+    for (let i of this.displayTraces) {
+      const length = (this.lastData) ? data.length + 1 : data.length;
+      const positions = new Float32Array(length * 3);
+      const dataPoints = new Float64Array(length * 2);
+
+      let offset = 0;
+
+      // Add the last point of data if it exists to avoid gaps.
+      if (this.lastData) {
+        dataPoints[offset] = this.lastData[0];
+        dataPoints[offset + 1] = this.lastData[i+1];
+        offset += 2;
+      }
+
+      for (let row of data) {
+        dataPoints[offset] = row[0];
+        dataPoints[offset + 1] = row[i + 1];
+        offset += 2;
+      }
+
+      if (dataPoints.length > 1) {
+        const geometry = new THREE.BufferGeometry();
+        geometry.addAttribute('position', new THREE.BufferAttribute(positions, 3));
+        geometry.addAttribute('data', new THREE.BufferAttribute(dataPoints, 2));
+
+        let material = new THREE.LineBasicMaterial({
+          color: COLOR_LIST[i % COLOR_LIST.length],
+          linewidth: 1.5
+        });
+        let mesh = new THREE.Line(geometry, material);
```

Comment:
These last two could use `const` instead. Also, maybe call the latter `line` instead of `mesh`.